src/agreementManager/AgreementManager.ts

```diff
@@ -77,45 +111,191 @@ class AgreementManager {
             throw new Error(
                 "AgreementManager - confirmBlock - block already confirmed"
             );
+        agreement.blockConfirmation.signatures.push(
+            confirmationSignature as BytesLike
+        );
+    }
+
+    public getForkGenesisSnapshot(
+        forkCnt: number
+    ): StateSnapshotStruct | undefined {
+        // TODO:
+        return undefined;
+    }
+
+    public getJoinChannelChain(): JoinChannelBlockStruct[] {
+        const blocks: JoinChannelBlockStruct[] = [];
+        const channelBlocks = this.joinChannelBlocks;
+
+        if (!channelBlocks) return blocks;
+
+        let currentHash = this.latestJoinChannelBlockHash;
+        if (!currentHash) return blocks;
+
+        while (true) {
+            const block = channelBlocks.get(currentHash);
+            if (!block) {
+                throw new Error(`Chain is broken at hash ${currentHash}`);
+            }
+
+            blocks.unshift(block);
+
+            // Check if this is the genesis block (block's hash equals its previousBlockHash)
+            const blockHash = ethers.keccak256(
+                EvmUtils.encodeJoinChannelBlock(block)
```

Comment:
encodeBlock not encodeJoinChannelBlock (in new code which is not here it's just 'encode')